axum/src/extract/host.rs

```diff
@@ -130,8 +141,14 @@ mod tests {
 
     #[crate::test]
     async fn uri_host() {
-        let host = test_client().get("/").await.text().await;
-        assert!(host.contains("127.0.0.1"));
+        let mut parts = Request::new(()).into_parts().0;
+        parts.uri = "https://127.0.0.1:1234/image.jpg".parse().unwrap();
+        let host = Host::from_request_parts(&mut parts, &()).await.unwrap();
+        assert_eq!(host.0, "127.0.0.1:1234");
+
+        parts.uri = "http://cool:user@[::1]:456/file.txt".parse().unwrap();
+        let host = Host::from_request_parts(&mut parts, &()).await.unwrap();
+        assert_eq!(host.0, "[::1]:456");
```

Comment:
Why did you completely rewrite this example to not use the test client?

Reply:
I've re-introduced the test client in https://github.com/tokio-rs/axum/pull/2242/commits/e4a109ed5a713b100d6af07c6f75c392eb504df8